pyomo/contrib/alternative_solutions/solnpool.py

```diff
@@ -9,119 +9,1004 @@
+        self._heap = []
+        self._unique_solutions = set()
+
+    @property
+    def pool_config(self):
+        return dict(
+            max_pool_size=self.max_pool_size,
+            objective=self.objective,
+            abs_tolerance=self.abs_tolerance,
+            rel_tolerance=self.rel_tolerance,
+            sense_is_min=self.sense_is_min,
+            best_value=self.best_value,
+        )
 
-    return solutions
+    def add(self, *args, **kwargs):
+        """
+        Add inputted solution to SolutionPool.
+        Relies on the instance as_solution conversion method to convert inputs to Solution Object.
+        If solution already present or outside tolerance of the best objective value, new solution is not added.
+        If input solution is new and within tolerance of the best objective value, the converted Solution object to the pool dictionary.
+        ID value for the solution generated as next increment of instance PoolCounter.
+        When pool size < max_pool_size, new solution is added without deleting old solutions.
+        When pool size == max_pool_size, new solution is added and oldest solution deleted.
+
+        Parameters
+        ----------
+        Input needs to match as_solution format from pool initialization.
+
+        Returns
+        ----------
+        None or int
+            None value corresponds to solution was already present and is ignored.
+            When not present, the ID value to match the added solution from the solution pool's PoolCounter.
+            The ID value is also the pool dictionary key for this solution.
+        """
+        soln = self._as_solution(*args, **kwargs)
+        #
+        # Return None if the solution has already been added to the pool
+        #
+        tuple_repn = soln._tuple_repn()
+        if tuple_repn in self._unique_solutions:
+            return None
+        self._unique_solutions.add(tuple_repn)
+        #
+        value = soln.objective(self.objective).value
+        keep = False
+        new_best_value = False
+        if self.best_value is nan:
+            self.best_value = value
+            keep = True
+        else:
+            diff = (
+                value - self.best_value
+                if self.sense_is_min
+                else self.best_value - value
+            )
+            if diff < 0.0:
+                # Keep if this is a new best value
+                self.best_value = value
+                keep = True
+                new_best_value = True
+            elif ((self.abs_tolerance is None) or (diff <= self.abs_tolerance)) and (
+                (self.rel_tolerance is None)
+                or (
+                    diff / min(math.fabs(value), math.fabs(self.best_value))
+                    <= self.rel_tolerance
+                )
+            ):
+                # Keep if the absolute or relative difference with the best value is small enough
+                keep = True
+
+        if keep:
```

Comment:
Minor nit, but rather than defining and checking `keep`, I think you can just return `None` whenever you would set `keep` to `False` and then you don't need this if statement.